### DISCUSSION TITLE
How to install Breeze in restricted environment?

### DISCUSSION THREAD

**Discussion by vijaya-lakshmi-venkatraman** [apache/airflow#21755] [Q&A]
Hello All,
I am trying to install Breeze in our restricted environment (which cannot access external websites).
Please can you advise if there is a documentation on this?
If not, please can guide me how to change Breeze script so as to use the images from our local repo as l have been facing errors due to external access when I start running ./breeze from within our VM?

Should I start with building new docker image as specified in [this link](https://airflow.apache.org/docs/docker-stack/build.html#build-images-in-security-restricted-environments)?

**Comment by potiuk** [comment]
Breeze is not going to work for multiple reasons if you have restricted environment. Even if you try to build image, the docker build steps are often pulling stuff from various sources. 

My recommendation is - if you want to develop airflow, use a machine that is not restricted. There is no recipe for restricted environments and it is unlikely it will ever be.

**Comment by potiuk** [comment]
We are going to have `codespaces` enabled soon https://github.com/features/codespaces  - this will setup "breeze" development environment in GitHub Cloud and you will be able to use VSCode full power there. 

If you want to develop Airlfow from restricted network and you already have access to GitHub, you should simply ask your company for enabling codespaces for your account (or wait until it will be available for your private account) - but until you have the access setting up a machine outside of restricted network is definitely recommended way to go.
